### PR TITLE
Updated component documentation/Alert

### PR DESCRIPTION
**Pull request by heatherlarsen**
### :pushpin: Summary
If merged, this PR would update the documentation for the `alert` component. 

👉 👉 👉 Preview: https://hds-website-git-heatherlarsen-alert-docs-hashicorp.vercel.app/components/alert/

### :hammer_and_wrench: Detailed description
Lots of documentation updates here, but the biggest changes included:
- Moving design guidelines to the "Guidelines" tab
- Moving anatomy content to the "Specifications" tab
- Cleaning up/editing content for a more natural reading flow
- Aligning the order of the property sections across the tabs, as necessary
- Adding in alerts/banners, where appropriate
- Tried to break up the markdown files into logical chunks

Known issues/things to resolve in follow-up PRs:
- Wrapping issue in the sidecar for longer content (can be seen in the Accessibility tab)
- Image sizing (they feel a little large)
- Content spacing ([HDS-1081](https://hashicorp.atlassian.net/browse/HDS-1081))
- How do we want to handle the **Showcase** section? I created it manually. cc @didoo 
- Is there a way to add the component cards in the "Related" section? cc @didoo 
- UI polish on the WCAG component

### :camera_flash: Screenshots
Guideline documentation for alert
Code documentation for alert
Specification documentation for alert
Accessibility documentation for alert

### :link: External links
Jira ticket: [HDS-1205](https://hashicorp.atlassian.net/browse/HDS-1205)
[Figma file](https://www.figma.com/file/Ky0qWjvHZR3je1lCBlzNB1/HDS-website?node-id=1336%3A78952&t=P6NML3fbUC413y1l-4)

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: website/docs/components/alert/partials/specifications/anatomy.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## Anatomy
+
+### Page and inline alerts
+
+![Anatomy of the page alert](/assets/components/alert/alert-anatomy-inline.png)
+
+| Element          |                                                 |
```
Comment (jorytindall) [inline website/docs/components/alert/partials/specifications/anatomy.md:7]: This is probably a broader question (will be present in many other pages), but should there be a label in the column header? Probably a requirement for screen readers, could be something like "usage" or similar

File: website/docs/components/alert/partials/specifications/anatomy.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+## Anatomy
+
+### Page and inline alerts
+
+![Anatomy of the page alert](/assets/components/alert/alert-anatomy-inline.png)
```
Comment (jorytindall) [inline website/docs/components/alert/partials/specifications/anatomy.md:5]: [small nit] Should the text in the dismiss button label be on one line?

**Comment by didoo** [comment]
> * How do we want to handle the **Showcase** section? I created it manually. cc @didoo

@heatherlarsen (+ @Dhaulagiri) I would suggest to leave the "Showcase" sections as they are (unless they break the page) and we will re-consider them holistically in January (we have to understand how to use them in the context of the new documentation format).

> * Is there a way to add the component cards in the "Related" section? cc @didoo

@heatherlarsen the "related components" feature will be implemented in January, using a much simpler approach (a new field in the "frontmatter" of a markdown file)

**Review by MelSumner — APPROVED** [review]

File: website/docs/components/alert/partials/specifications/anatomy.md (hunk 1 above)
Comment (heatherlarsen) [reply website/docs/components/alert/partials/specifications/anatomy.md:7]: Yea, I realized that we can't do tables without headings in markdown, so I had to put something. I couldn't come up with anything for the second column, but I like "usage".

File: website/docs/components/alert/partials/specifications/anatomy.md (hunk 2 above)
Comment (heatherlarsen) [reply website/docs/components/alert/partials/specifications/anatomy.md:5]: Good catch. I'll fix it.

**Review by jorytindall — APPROVED** [review]
Thanks for the updates, looks good!

**Pull request merged**